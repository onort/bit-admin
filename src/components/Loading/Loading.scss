@import "../../styles/main.scss";

// credit: http://tobiasahlin.com/spinkit/

.container {
  margin: px-to-rem(100px) auto;
  width: 100%;
  text-align: center;
}

.fullscreen {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  margin: 0;
  padding: 20% 0 0;
  z-index: 9999;
  background-color: $lighter-bg;
}

[class*="_dot"] {
  width: $spacer-md;
  height: $spacer-md;
  margin-left: $spacer-xxsm;
  background-color: $text-color;
  border-radius: 100%;
  display: inline-block;
  animation: bouncedelay 1.4s infinite ease-in-out both;
}

.container > .dot1 {
  animation-delay: -0.32s;
}

.container > .dot2 {
  animation-delay: -0.16s;
}

.container > .dot3 {
  animation-delay: 0s;
}

@keyframes bouncedelay {
  0%,
  80%,
  100% {
    transform: scale(0);
  }
  40% {
    transform: scale(1);
  }
}
