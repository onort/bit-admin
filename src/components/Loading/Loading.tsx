import React from "react"
import cx from "classnames"

import styles from "./Loading.scss"

interface Props {
  className?: string
  dotClassName?: string
  fullscreen?: boolean
}

const Loading: React.SFC<Props> = props => {
  const className = cx(
    styles.container,
    { [styles.fullscreen]: props.fullscreen },
    props.className
  )
  return (
    <div className={className}>
      <div className={cx(styles.dot1, props.dotClassName)} />
      <div className={cx(styles.dot2, props.dotClassName)} />
      <div className={cx(styles.dot3, props.dotClassName)} />
    </div>
  )
}

Loading.defaultProps = {
  fullscreen: false
}

export default Loading
