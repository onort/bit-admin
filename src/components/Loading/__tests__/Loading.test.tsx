import React from "react"
import { shallow } from "enzyme"

import Loading from "../"

const testClassName = "testClassName"

describe("<Loading />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Loading />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should render fullscreen", () => {
    const wrapper = shallow(<Loading fullscreen={true} />)
    expect(wrapper.find(".container").hasClass("fullscreen")).toBe(true)
  })

  it("should have custom classname", () => {
    const wrapper = shallow(<Loading className={testClassName} />)
    expect(wrapper.find(".container").hasClass(testClassName)).toBe(true)
  })

  it("should have custom dot classname", () => {
    const wrapper = shallow(<Loading dotClassName={testClassName} />)
    expect(wrapper.find(".dot1").hasClass(testClassName)).toBe(true)
  })
})
