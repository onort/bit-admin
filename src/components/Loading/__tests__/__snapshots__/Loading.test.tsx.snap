// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="dot1"
  />
  <div
    className="dot2"
  />
  <div
    className="dot3"
  />
</div>
`;

exports[`<Loading /> should match snapshot 2`] = `"<div class=\\"container\\"><div class=\\"dot1\\"></div><div class=\\"dot2\\"></div><div class=\\"dot3\\"></div></div>"`;
